Include/internal/pycore_optimizer.h

```diff
@@ -220,15 +221,69 @@ typedef union _jit_opt_symbol {
 } JitOptSymbol;
 
 
+// This mimics the _PyStackRef API
+typedef union {
+    uintptr_t bits;
+} JitOptRef;
+
+#define JIT_BITS_TO_PTR(REF) ((JitOptSymbol *)((REF).bits))
+#define JIT_BITS_TO_PTR_MASKED(REF) ((JitOptSymbol *)(((REF).bits) & (~Py_TAG_REFCNT)))
+
+static inline JitOptSymbol *
+PyJitRef_AsSymbolBorrow(JitOptRef ref)
+{
+    return JIT_BITS_TO_PTR_MASKED(ref);
+}
+
+bool _Py_uop_sym_is_immortal(JitOptSymbol *sym);
+
+static inline JitOptRef
+PyJitRef_FromSymbolSteal(JitOptSymbol *sym)
+{
+    if (sym == NULL || _Py_uop_sym_is_immortal(sym)) {
+        return (JitOptRef){.bits=(uintptr_t)sym | Py_TAG_REFCNT};
```

Comment:
just to avoid some duplication (but the way it is now is also fine, so feel free to disregard :))
```suggestion
        return PyJitRef_FromSymbolBorrow(sym);
```